### ISSUE TITLE
Left-side API list, contains private classes

### ISSUE THREAD

**Issue by stefanpenner** [emberjs/website#2452]

**Comment by acorncom** [comment]
Per a discussion of this at last week's docs team meeting @toddjordan was going to look into fixing this ...

**Comment by toddjordan** [comment]
This week all take a look and try to list here the actual classes that need removal.  Feel free if anyone knows of specific classes that shouldn't be documented to comment.

**Comment by stefanpenner** [comment]
unfortunately likely checking each link, and seeing if PRIVATE is mentioned in the corresponding page description.

example: http://emberjs.com/api/classes/ContainerProxyMixin.html contains PRIVATE, but still exists in Classes.

If required, I can go through this manually, feel free to let me know.

**Comment by toddjordan** [comment]
np, I can do it.

**Comment by rwjblue** [comment]
We should also suppress deprecated classes.

**Comment by toddjordan** [comment]
Alright, compiled a list of what needs to be trimmed. The only way I know to remove so yuidoc doesn't put it in the left nav is to remove the doc comments for the class on each of the files.  Is this right, or is there a better way?

Also, the following classes have no documentation and no private/public indication.  Should I also remove?
HTMLBars
RSVP.EventTarget
String

**Comment by toddjordan** [comment]
looks like @krisselden took this up on the ruby side of things...  I'll move on to then next thing on the list then.

**Comment by toddjordan** [comment]
I was looking on the other side of things - trying out some things on the yuidoc side to limit the output to exclude the privates/deprecations.  I might just round out some of what I was trying there before completely moving on, but it may be that the approach here is the way to go.

**Comment by toddjordan** [comment]
So I did verify that removing the @class line for each private/deprecated class also will keep the classes from showing in docs.  So taking this approach will just be deleting a line for each private and deprecated class.  I don't know if we want to go this route or continuing to generate docs for everything.  

One argument for removing @class would be that I don't think we want private classes showing in api docs, and there's multiple clients showing these (for example: http://devdocs.io/ember/ )

**Comment by toddjordan** [comment]
Moving away from just removing `@class` since it doesn't keep privates from being added back.  I'm trying out updating ember-cli-yuidoc (used by ember) to use something like yuidoc-filter-tags as a preprocessor for the option to filter tags with `@private` and `@deprecated`

**Comment by toddjordan** [comment]
Created a PR to ember to add an npm package I created to filter out private and deprecated classes prior to render.

**Comment by toddjordan** [comment]
After some discussion on how to address this issue in PR https://github.com/emberjs/ember.js/issues/13075 we have decided to go forward with the following proposal as worked out between @wycats and @Gaurav0 

From @wycats in the PR: 

There are already existing links that point to private and deprecated APIs on Stack Overflow etc. We should try not to break them.
1. Deprecated APIs are still conceptually part of the official semver API, and shouldn't be lumped in with "private" APIs.
2. A number of private classes have public methods. A notable example of this is Ember.View. If we simply removed Ember.View from the docs, much of the current documentation for Ember.Component would disappear.
3. Linking to source code for private APIs works as a solution for linking to the docs for that precise API, but it's difficult (sometimes extremely difficult) to follow @see cross-links.

The rough plan of action, in order, seems to be:

1. Create a new internals subdomain that is a mirror of the api subdomain. We'd use <link rel="canonical"> on the internals subdomain to avoid google pointing people to internals pages that have an equivalent on api.
2. Create toggles to hide (but not remove) deprecated and private APIs. The internals subdomain defaults to "on", while the api subdomain defaults to "off".
3. Add an automatic redirect from private APIs to the internals subdomain. This redirect would be emitted together with the code that knows how to hide the private APIs. For methods, we'd detect the #privateMethod URL hash and do a dynamic redirect.
4. Once this is done, we'd remove private APIs completely from the api, but leave the redirects.
